examples/source_examples/accelo_api_connector_multithreading_enabled/connector.py

```diff
@@ -0,0 +1,828 @@
+
+        update_start_time = time.time()
+
+        for entity_sync in [sync_companies, sync_invoices, sync_payments, sync_prospects, sync_jobs, sync_staff]:
+            entity_name = entity_sync.__name__.replace("sync_", "")
+            yield from entity_sync(access_token)
+
+        yield op.checkpoint(thread_local_state.state)
+
+        update_duration = time.time() - update_start_time
+        log.info(
+            f"Update process completed. Total time: {update_duration:.2f} seconds. Final state: {thread_local_state.state}")
+    except Exception as e:
+        log.severe(f"Update process failed: {str(e)}")
+
+
+def sync_entity(entity_name, access_token, fields, last_sync_key, process_record=None, date_field="date_modified",
+                timeout=SYNC_TIMEOUT, batch_size=BATCH_SIZE, fetch_data_func=None):
+    """
+    Sync data for a specific entity from Accelo API.
+
+    This function handles the core logic for syncing each entity type:
+    - Fetches data in batches
+    - Processes records
+    - Yields upsert operations
+    - Updates the sync state
+
+    Args:
+        entity_name (str): The name of the entity being synced.
+        access_token (str): The OAuth 2.0 access token.
+        fields (list): The fields to sync for this entity.
+        last_sync_key (str): The key to use for storing the last sync time in the state.
+        process_record (callable, optional): A function to process each record before upserting.
+        date_field (str, optional): The field to use for incremental syncing.
+        timeout (int, optional): Timeout for the entire sync operation in seconds.
+        batch_size (int, optional): Number of records to fetch per API request.
+        fetch_data_func (callable, optional): A custom function to fetch data for this entity.
+
+    Yields:
+        dict: Upsert operations for each synced record.
+    """
+    log.info(f"Starting sync for {entity_name}")
+    try:
+        last_sync = thread_local_state.state.get(last_sync_key, "1970-01-01T00:00:00Z")
+        last_sync_unix = max(0, int(datetime.fromisoformat(last_sync.replace("Z", "+00:00")).timestamp()))
+        log.info(f"Last sync time for {entity_name}: {last_sync} (Unix: {last_sync_unix})")
+
+        params = {
+            "_fields": ",".join(fields + [date_field]),
+            "_filters": f"{date_field}_after({last_sync_unix})",
+            "_limit": batch_size,
+            "_page": 0,
+            "_order_by": date_field
+        }
+
+        records_processed = 0
+        entities_received = 0
+        start_time = time.time()
+        max_date_value = last_sync_unix
+
+        def fetch_page(page):
+            nonlocal entities_received, max_date_value
+            current_params = params.copy()
+            current_params['_page'] = page
+            try:
+                entities = fetch_data_func(current_params) if fetch_data_func else fetch_data(entity_name, access_token,
```

Comment:
Is there any place we provide `fetch_data_func`, I think we can directly use `fetch_data`, please confirm and remove it.

Reply:
Yes it's not used anywhere else, but I think its to keep it customisable, not sure so I think we can let it be as it's minor.